io/plugins/eu.esdihumboldt.hale.io.haleconnect/src/eu/esdihumboldt/hale/io/haleconnect/HaleConnectService.java

```diff
@@ -138,4 +185,70 @@
 	LocatableInputSupplier<InputStream> loadProject(Owner owner, String projectId)
 			throws HaleConnectException;
 
+	/**
+	 * Create a new transformation project
+	 * 
+	 * @param name Project name
+	 * @param author Project author
+	 * @param owner Project owner
+	 * @param versionControl whether to activate version control for the project
+	 * @return the project ID
+	 * @throws HaleConnectException thrown on any API exception
+	 */
+	String createProject(String name, String author, Owner owner, boolean versionControl)
+			throws HaleConnectException;
+
+	/**
+	 * Upload a project file
+	 * 
+	 * @param projectId Transformation project ID
+	 * @param owner Project owner
+	 * @param file the file to upload
+	 * @param progress a progress indicator
+	 * @return true if the upload was successful
+	 * @throws HaleConnectException thrown on any API exception
+	 */
+	boolean uploadProjectFile(String projectId, Owner owner, File file, ProgressIndicator progress)
+			throws HaleConnectException;
+
+	/**
+	 * Upload a project asynchronously
+	 * 
+	 * @param projectId Transformation project ID
+	 * @param owner Project owner
+	 * @param file the file to upload
+	 * @param progress a progress indicator
+	 * @return a {@link ListenableFuture} whose value will be set to true upon
+	 *         successful completion of the request or false when the request
+	 *         fails
+	 * @throws HaleConnectException thrown on any API exception
+	 */
+	ListenableFuture<Boolean> uploadProjectFileAync(String projectId, Owner owner, File file,
```

Comment:
The method actually uploads only a single file (which may be an archive though).